bin/nim-gdb

```diff
@@ -5,17 +5,25 @@ set -e
 
 which nim > /dev/null || (echo "nim not in PATH"; exit 1)
 which gdb > /dev/null || (echo "gdb not in PATH"; exit 1)
+which readlink > /dev/null || which greadlink > /dev/null || (echo "readlink not in PATH. Please install coreutils from brew if on Mac."; exit 1)
+
+nreadlink () {
+    (which greadlink > /dev/null && greadlink "$@") || (which readlink > /dev/null && readlink "$@") || echo "Readlink could not be found"
+}
 
 # Find out where the pretty printer Python module is
-NIM_SYSROOT=$(dirname $(dirname $(readlink -e $(which nim))))
+NIM_SYSROOT=$(dirname $(dirname $(nreadlink -e $(which nim))))
 GDB_PYTHON_MODULE_PATH="$NIM_SYSROOT/tools/nim-gdb.py"
 
 # Run GDB with the additional arguments that load the pretty printers
 # Set the environment variable `NIM_GDB` to overwrite the call to a
 # different/specific command (defaults to `gdb`).
 NIM_GDB="${NIM_GDB:-gdb}"
+
+# This is needed for some reason. I can't get -eval-command to work ever
+echo "source $GDB_PYTHON_MODULE_PATH" > gdbcommands.txt
 # exec replaces the new process of bash with gdb. It is always good to
 # have fewer processes.
-exec ${NIM_GDB} \
-  -eval-command "source $GDB_PYTHON_MODULE_PATH" \
-  "$@"
+exec ${NIM_GDB} --command="gdbcommands.txt" "$@"
+
+rm gdbcommands.txt
```

Comment:
this line will never be executed. Nothing after ``exec`` will be executed because like I explained in the comment above, ``exec`` replaces the process of bash with gdb. There is no ``bash`` precess running anymore after ``exec``.

Reply:
Yep, you're right. Don't know why I thought this worked. I'll open a PR and fix the issues you mentioned here and add you as a reviewer